core/src/main/java/eu/fasten/core/data/JSONUtils.java

```diff
@@ -0,0 +1,287 @@
+package eu.fasten.core.data;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class JSONUtils {
+    /**
+     * Converts an {@link ExtendedRevisionJavaCallGraph} object to its corresponding JSON String
+     * without any object creation in between. It creates a {@link StringBuilder) in the beginning
+     * and only appends to it in order to decrease the memory and time consumption.
+     *
+     * @param erjcg and object of java revision call graph to be converted to JSON String.
+     * @return the corresponding JSON String.
+     */
+    public static String toJSONString(final ExtendedRevisionJavaCallGraph erjcg) {
+        var result = new StringBuilder("{");
+        appendArtifactInformation(erjcg, result);
+        appendCha(result, erjcg.classHierarchy);
+        appendGraph(result, erjcg.graph);
+        if (erjcg.timestamp >= 0) {
```

Comment:
In case timestamp value is negative (probably not a valid scenario) we will have a trailing comma coming from `appendGraph`call.

JSON Java accept the JSON with the trailing comma whereas JSON specification don't (see https://github.com/stleary/JSON-java/issues/269#issuecomment-239738561) so I think this is not a real issue.

I'm reporting this because we don't get strictly the same result as with the previous implementation.

Reply:
fixed afdb9bf